.github/workflows/docs.yaml

```diff
@@ -0,0 +1,60 @@
+name: build docs
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release/*'
+    tags:
+      - 'v*'
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8]
+    env:
+      DISPLAY: ':99.0'
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: setup ubuntu-latest xvfb
+      uses: ./.github/actions/setup-ubuntu-latest-xvfb
+    - name: install pandoc ubuntu
+      run: sudo apt install pandoc
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: prepare pip cache
+      id: prepare-cache-pip
+      uses: ./.github/actions/prepare-cache-pip
+    - name: pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/docs_reqirements.txt') }}-${{ hashFiles('**/setup.py') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    - name: install requirements.txt
+      run: |
+        pip install -r requirements.txt --use-feature=2020-resolver
+    - name: install docs_requirements.txt
+      run: |
+        pip install -r docs_requirements.txt --use-feature=2020-resolver
+    - name: install qcodes
+      run: pip install . --use-feature=2020-resolver
```

Comment:
is it a useful idea to extract common steps across the "workflows" into some kind of "install ubunu, python, pip, qcodes minimum" "step grouping" thing? is that even possible? azure pipelines had these "templates", smth where the pipeline can take steps from

Reply:
As far as I understand it is possbile to create composable actions of one or more run stesps but not to put actions into other composable actions. This limits a bit what we can do. E.g. that is why the